gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1603,12 +1612,45 @@ gst_tensor_src_iio_get_times (GstBaseSrc * basesrc, GstBuffer * buffer,
 
 /**
  * @brief create a buffer with requested size and offset
+ * @note offset ignored as there is no header
+ * @note size ignored as done in pushsrc
  */
 static GstFlowReturn
 gst_tensor_src_iio_create (GstBaseSrc * src, guint64 offset,
     guint size, GstBuffer ** buffer)
 {
-  /** FIXME: fill this function */
+  GstTensorSrcIIO *self;
+  GstBuffer *buf;
+  GstMemory *mem;
+  GstMapInfo info;
+  guint buffer_size;
+
+  self = GST_TENSOR_SRC_IIO_CAST (src);
+  buffer_size = gst_tensor_info_get_size (&self->tensors_config->info.info[0]);
+
+  mem = gst_allocator_alloc (NULL, buffer_size, NULL);
+  if (!gst_memory_map (mem, &info, GST_MAP_WRITE)) {
+    goto error_mem_free;
+  }
+  info.size = size;
```

Comment:
In GstMapInfo info; > info.size
Is it necessary? I think we don't need to do gst_memory_map() - gst_memory_unmap()
IMO, just check mem is not null in line 1632.

Reply:
Or check buffer_size == size

Reply:
Added memory check for null
Added bug fix - the function arguments `offset` and `size` should be ignored as they are for pull mode.
`info.size' should be set with buffer_size - and yes, gst_memory_map() - gst_memory_unmap() is reduntant.
updated.